articles/you-are-not-a-software-engineer.md

```diff
@@ -0,0 +1,58 @@
+
+## "But I am a coder, I don't want to deal with any of this."
+
+Yes, I understand and I am too. Coding is fun, coding can be challenging, coding is the reason most of embarked on this journey. But if you think about it, code is made for the human consumption first. Yes, machines need to understand it as well, but if it wasn't for the people, we would still write in assembly, or binaries even. When you throw in daily collaboration with your team, managers, and clients, your job is really about dealing with people, not computers.
+
+Let me put it this way. You may put in the work and produce 10.000 lines of cleanest, most optimized code there possibly is. If it doesn't bring value to the business, you can go ahead and throw it away. You can create a fresh, shiny feature, which works as a charm. The downside is, it takes 20 iterations, because of no deep understanding of what the customer really needs. And maybe after the second iteration, you could tell that customer need would be fulfilled with an entirely different feature.
+
+## What is your drive?
+
+You don't want to just play around and not produce anything of worth to someone, I hope.
+
+According to recent models of motivation ("Drive" by Daniel Pink), our actions are fueled from autonomy, mastery, and purpose.
+
+In healthy working environments, autonomy is easy. You have the power to decide how will you implement the solution, to choose working mode which enables you to do so, and influence decisions, within your ability.
+
+Mastery is the urge to improve your skill set and be proud after all the work you have done. This moves us. To change projects, gather new experiences and work with new technologies. Since IT field is a fast-moving freight train, I assume you are already doing this.
+
+Purpose can be tricky, though. The desire to do something that is of meaning and importance, greater than ourselves. Maybe helping already successful enterprise earn 5% net profit greater than previous year is not your thing. Maybe you want to help humanity find the cure for cancer. Boarding on some ideas is easy, while others not. I will not provide an answer to this. It is entirely up to you. I will give you a hint though. It is everywhere. It involves feelings.
+
+## "I am in the business of creating value."
+
+Remember the cost of fixing a bug, depending on the stage of software development lifecycle. An hour spent in requirements engineering will save you many more down the road. You might use a framework designed to prevent this from happening, prescribing you the points in time when you should do requirements engineering. But, all the magical frameworks in the world will not help you, if you don't understand what you are building.
```

Comment:
In diesem Abschnitt sehe ich die Aussage zu wenig. Auf was genau soll der Leser künfig achten / was soll er anders machen?

Reply:
In this paragraph I do not see the statement you are making? What should the reader pay attention to in the future or what should she do differently?

Reply:
I have multiple points:
- we need to understand what we are building and not be input requirements/output code machine
- using an agile framework doesn't grant you value generation per se
- making sure code is creating value is our responsibility (of the person doing the act of coding)

Apparently the text does get the message through. Or you don't agree with the message?

Let me try and refactor it a bit.